modin/pandas/series.py

```diff
@@ -719,7 +719,9 @@ def between(self, left, right, inclusive: str = "both"):  # noqa: PR01, RT01, D2
         Return boolean Series equivalent to left <= series <= right.
         """
         return self.__constructor__(
-            query_compiler=self._query_compiler.between(left, right, inclusive)
+            query_compiler=self._query_compiler.between(
+                left=left, right=right, inclusive=inclusive
+            )
         )
```

Comment:
it looks like the logic of this method can be easily implemented at the front end, I believe we don't need to add this new qc method.

We can either copy the [pandas' logic](https://github.com/pandas-dev/pandas/blob/e86ed377639948c64c429059127bcf5b359ab6be/pandas/core/series.py#L5424-L5435) or just pass our own object to the pandas method:
```python
def between(self, left, right, inclusive):
    # 'pandas.between()' only uses public Series' API, so passing a Modin Series there is safe
    return pandas.Series.between(self, left, right, inclusive)
```